### PR TITLE
DB-10253 Extend AS OF to all types of scan

### PR DESCRIPTION
**Pull request by hatyo**

### CODE REVIEW

**Review by yxia92 — APPROVED** [review]
+1

**Comment by hatyo** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/1543/

**Comment by hatyo** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/1550/

**Comment by hatyo** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

File: db-engine/src/main/java/com/splicemachine/db/iapi/sql/dictionary/TableDescriptor.java [hunk 1]
```diff
@@ -399,11 +399,11 @@ public void setPurgeDeletedRows(boolean purgeDeletedRows) {
         this.purgeDeletedRows = purgeDeletedRows;
     }
 
-    public Long getMinRetainedVersions() {
+    public Long minRetainedPeriod() {
```
Comment (ipraznik-splice) [inline db-engine/src/main/java/com/splicemachine/db/iapi/sql/dictionary/TableDescriptor.java:402]: should be here maybe getMinRetainedPeriod to use a standard getter style?

File: splice_machine/src/main/java/com/splicemachine/derby/impl/sql/catalog/SpliceDataDictionary.java [hunk 2]
```diff
@@ -1666,6 +1669,11 @@ public void addMinRetentionPeriodColumn(TransactionController tc) throws Standar
             DataTypeDescriptor dtd = DataTypeDescriptor.getBuiltInDataTypeDescriptor(Types.BIGINT, 1);
             tc.addColumnToConglomerate(td.getHeapConglomerateId(), colNumber, storableDV, dtd.getCollationType());
 
+            try {
+                Thread.sleep(10000);
```
Comment (ipraznik-splice) [inline splice_machine/src/main/java/com/splicemachine/derby/impl/sql/catalog/SpliceDataDictionary.java:1673]: not sure how often addMinRetentionPeriodColumn is called, but to have every 10 secs delay quite expensive. Why is it needed here?

**Review by ipraznik-splice — CHANGES_REQUESTED** [review]
please look at my comments

**Comment by spliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/1553/

File: db-engine/src/main/java/com/splicemachine/db/iapi/sql/dictionary/TableDescriptor.java (hunk 1 above)
Comment (hatyo) [reply db-engine/src/main/java/com/splicemachine/db/iapi/sql/dictionary/TableDescriptor.java:402]: done.

**Comment by hatyo** [comment]
Thanks Igor for reviewing the PR, I addressed your comments

**Review by ipraznik-splice — APPROVED** [review]

**Review by carolp-503 — APPROVED** [review]
+1

**Review by jyuanca — APPROVED** [review]

**Comment by hatyo** [comment]
Will recreate this PR on top on master instead of DB-10084.

**Comment by hatyo** [comment]
Moved [here](https://github.com/splicemachine/spliceengine/pull/4141).

**Pull request closed without merging**